regression/goto-harness/do-not-use-nondet-for-primitives/test.desc

```diff
@@ -0,0 +1,17 @@
+CORE
+test.c
+--function test_function --harness-type call-function
+\[test_function\.assertion\.1\] line \d+ assertion test != 42
+^EXIT=10$
+^SIGNAL=0$
+--
+nondet_signed_int
+--
+If we use side_effect_expr_nondett then dump_c will generation function calls
+to functions with names like nondet_signed_int without declaration.
+This can cause problems around usability.
+
+To fix this, we changed goto-harness to just not use side_effect_expr_nondett
+at all.
+
+This test tests for the absence the use for these things.
```

Comment:
:pick: This seems like great info for a commit message (since it is talking about why we moved from one solution to another) but in a test, where the reader won't have the context that we used to use `side_effect_expr_nondett` it is maybe confusing. I'd focus instead on what the test is testing - in this case not using undeclared functions.

Reply:
I note this info _is_ in the commit too :+1: I still think probably more confusing than helpful in the test, but up to you